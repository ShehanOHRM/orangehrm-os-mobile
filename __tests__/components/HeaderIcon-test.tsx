/*
 * This file is part of OrangeHRM
 *
 * Copyright (C) 2020 onwards OrangeHRM (https://www.orangehrm.com/)
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import 'react-native';
import React from 'react';
import HeaderIcon from 'components/HeaderIcon';
import {Provider} from 'react-redux';
import configureStore from 'store/configureStore';
import {render} from 'react-native-testing-library';
const mockStore = configureStore();

describe('components/HeaderIcon', () => {
  test('test HeaderIcon component', () => {
    const onPressMock = jest.fn();
    const node = render(
      <Provider store={mockStore}>
        <HeaderIcon name={'close'} onPress={onPressMock} />
      </Provider>,
    ).toJSON();
    expect(node).toMatchSnapshot();
  });
});
