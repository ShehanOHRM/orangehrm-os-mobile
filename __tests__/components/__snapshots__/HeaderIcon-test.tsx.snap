// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HeaderIcon test HeaderIcon component 1`] = `
<View
  accessible={true}
  focusable={true}
  forwardedRef={[Function]}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 0,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#007aff",
            "fontSize": 24,
            "marginLeft": 16,
            "marginRight": 16,
            "paddingTop": 2,
          },
          Object {
            "color": "#818181",
            "fontSize": 20,
          },
          Object {
            "color": "#ffffff",
            "fontSize": 24,
          },
        ],
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    type="MaterialCommunityIcons"
  >
    ó°…–
  </Text>
</View>
`;
